common/display/display.cpp

```diff
@@ -164,8 +164,18 @@ bool Display::GetActiveConfig(uint32_t *config) {
   return true;
 }
 
-bool Display::SetDpmsMode(uint32_t dpms_mode) {
-  return display_queue_->SetDpmsMode(dpms_mode);
+bool Display::SetDpmsMode(uint32_t power_mode) {
+
+  power_mode_ = power_mode;
+  uint32_t dpms_prop = display_queue_->GetDpmsProp();
+
+  if (power_mode == DisplayPowerMode::kOff ||
```

Comment:
As we discussed offline, we should move this to displayqueue and do it synchronously there, is mode is off or doze than suspend thread.